File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java
package de.tum.in.tumcampusapp.component.tumui.calendar;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import de.tum.in.tumcampusapp.R;
import de.tum.in.tumcampusapp.component.other.generic.activity.ActivityForAccessingTumOnline;
import de.tum.in.tumcampusapp.component.tumui.calendar.model.EventsResponse;
import de.tum.in.tumcampusapp.utils.Const;

/**
 * Activity showing the user's calendar. Calendar items (events) are fetched from TUMOnline and displayed as blocks on a timeline.
 */
public class CalendarActivity extends ActivityForAccessingTumOnline<EventsResponse> {

    public CalendarActivity() {
        super(R.layout.activity_calendar);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            long showDate = getIntent().getLongExtra(Const.EVENT_TIME, -1);
            String eventId = getIntent().getStringExtra(Const.KEY_EVENT_ID);
            Fragment fragment = CalendarFragment.newInstance(showDate, eventId);

            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.contentFrame, fragment)
                    .commit();
        }
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.kt
package de.tum.`in`.tumcampusapp.component.tumui.calendar

import android.os.Bundle

import de.tum.`in`.tumcampusapp.R
import de.tum.`in`.tumcampusapp.component.other.generic.activity.ActivityForAccessingTumOnline
import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.EventsResponse
import de.tum.`in`.tumcampusapp.utils.Const

/**
 * Activity showing the user's calendar. Calendar items (events) are fetched from TUMOnline and displayed as blocks on a timeline.
 */
class CalendarActivity : ActivityForAccessingTumOnline<EventsResponse>(R.layout.activity_calendar) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            val showDate = intent.getLongExtra(Const.EVENT_TIME, -1)
            val eventId = intent.getStringExtra(Const.KEY_EVENT_ID)
            val fragment = CalendarFragment.newInstance(showDate, eventId)

            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.contentFrame, fragment)
                    .commit()
        }
    }
}
